hazelcast/src/main/java/com/hazelcast/internal/partition/operation/PartitionReplicaSyncRequestOffloadable.java

```diff
@@ -214,30 +215,42 @@ public void start() throws Exception {
             try {
                 nodeEngine.getExecutionService().execute(ExecutionService.ASYNC_EXECUTOR,
                         () -> {
-                            // set partition as migrating to disable mutating
-                            // operations while preparing replication operations
-                            if (!trySetMigratingFlag()) {
-                                sendRetryResponse();
-                            }
-
                             try {
-                                Integer permits = getPermits();
-                                if (permits == null) {
-                                    return;
-                                }
-                                sendOperationsForNamespaces(permits);
-                                // send retry response for remaining namespaces
-                                if (!namespaces.isEmpty()) {
-                                    logNotEnoughPermits();
+                                // set partition as migrating to disable mutating
+                                // operations while preparing replication operations
+                                if (!trySetMigratingFlag()) {
                                     sendRetryResponse();
                                 }
+
+                                try {
+                                    Integer permits = getPermits();
+                                    if (permits == null) {
+                                        return;
+                                    }
+                                    sendOperationsForNamespaces(permits);
+                                    // send retry response for remaining namespaces
+                                    if (!namespaces.isEmpty()) {
+                                        logNotEnoughPermits();
+                                        sendRetryResponse();
+                                    }
+                                } finally {
+                                    clearMigratingFlag();
+                                }
                             } finally {
-                                clearMigratingFlag();
+                                sendResponse(null);
                             }
                         });
             } catch (RejectedExecutionException e) {
                 // if execution on async executor was rejected, then send retry response
-                sendRetryResponse();
+                try {
+                    sendRetryResponse();
+                } finally {
+                    sendResponse(null);
+                }
+            } catch (Throwable t) {
```

Comment:
nit: Regarding thrown exceptions, they are replied OperationRunnerImpl.handleOperationError() -> sendResponseAfterOperationError() so I don't think we need to catch throwable here.

Reply:
done.